README.md
<h1 align="center">
  PyQt5 Authentification App
</h1>

<h3 align="center">
  PyQt5 app with the possibility of registration and logging in by <a href="https://github.com/codelao">Code Lao</a>
</h3>

<p align="center">
  <a href="#requirements">
    Requirements
  </a> •
  <a href="#usage">
    Usage
  </a>
</p>


## Screenshots
<p>
  <img src="./example_pic.jpg" width="50%">
  <img src="./registration_pic.jpg" width="50%">
  <img src="./profile_pic.jpg" width="50%">
</p>


## Requirements
<p>
  <img src="https://img.shields.io/badge/python-red">
  <img src="https://img.shields.io/badge/PyQt5-blue">
</p>


## Usage
- App runs from [first_window.py](https://github.com/codelao/PyQt5-Authentification-App/blob/main/code/first_window.py) file
- This is an old project that I decided to finish


CHANGELOG.md
# 1.0.1
*2023-02-13*

### Added
- Auto database file creation after launching the app


# 1.0.0
*2023-02-13*

### Hi


code/db.py
import sqlite3


class Database:
    def __init__(self, db_file):
        self.con = sqlite3.connect(db_file, check_same_thread=False)
        self.con.cursor().execute("CREATE TABLE IF NOT EXISTS data(username TEXT PRIMARY KEY, password TEXT, join_date TEXT)")
        
    def add_userdata(self, username, password, join_date):
        with self.con:
            self.con.cursor().execute("INSERT INTO data VALUES (?, ?, ?)", (username, password, join_date,))

    def check_user(self, username):
        with self.con:
            result = self.con.cursor().execute("SELECT * FROM data WHERE username = ?", (username,)).fetchall()  
            return bool(result)

    def get_user_join_date(self, username):
        with self.con:
            return self.con.cursor().execute("SELECT join_date FROM data WHERE username = ?", (username,)).fetchone()[0]

    def get_user_password(self, username):
        with self.con:
            return self.con.cursor().execute("SELECT password FROM data WHERE username = ?", (username,)).fetchone()[0]

    def delete_user(self, username):
        with self.con:
            self.con.cursor().execute("DELETE FROM data WHERE username = ?", (username,))


code/join_window.py
import datetime
from login_window import *
from PyQt5.QtCore import Qt
from PyQt5.QtWidgets import QWidget, QLabel, QVBoxLayout, QPushButton, QLineEdit, QMessageBox
from PyQt5 import QtGui
from db import Database


class join_win(QWidget):
    def __init__(self):
        super().__init__()
        self.appear()
        self.UI()
        self.connections()

    def appear(self):
        self.setWindowTitle('Registration')
        self.setWindowFlags(Qt.WindowStaysOnTopHint)
        self.setStyleSheet('''
        QWidget{
            background-color: #04202C
        }
        QLabel{
            color: #C9D1C8;
            font: bold 20px
        }
        QPushButton{
            color: #C9D1C8;
            font: bold 14px;
            border-style: outset;
            border-width: 2px;
            border-radius: 10px;
            border-color: #5B7065;
            font: bold 14px;
            min-width: 10em;
            padding: 6px
        }
        QPushButton:pressed{
            background-color: #5B7065
        }
        QLineEdit{
            border: 5px solid #5B7065;
            font: bold 20px;
            color: #C9D1C8
        }
        '''
        )
        self.resize(700, 450)
        self.move(400, 300)
        self.setWindowIcon(QtGui.QIcon('logo.png'))

    def UI(self):
        self.username_text = QLabel('Username:')
        self.username = QLineEdit()
        self.password_text = QLabel('Password:')
        self.password = QLineEdit()
        self.repeat_password_text = QLabel('Repeat password:')
        self.repeat_password = QLineEdit()
        self.join_button = QPushButton('Join')

        self.password.setEchoMode(QLineEdit.Password)
        self.repeat_password.setEchoMode(QLineEdit.Password)

        self.line = QVBoxLayout()
        self.line.addWidget(self.username_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.username, alignment=Qt.AlignCenter)
        self.line.addWidget(self.password_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.password, alignment=Qt.AlignCenter)
        self.line.addWidget(self.repeat_password_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.repeat_password, alignment=Qt.AlignCenter)
        self.line.addWidget(self.join_button, alignment=Qt.AlignCenter)
        self.setLayout(self.line)

    def connections(self):
        self.join_button.clicked.connect(self.click)

    def click(self):
        self.stylesheet = '''
        QMessageBox{
            background-color: #04202C
        }
        QLabel{
            color: #C9D1C8;
            font: bold 20px
        }
        QPushButton{
            color: #C9D1C8;
            font: bold 14px;
            border-style: outset;
            border-width: 2px;
            border-radius: 10px;
            border-color: #5B7065;
            font: bold 14px;
            min-width: 10em;
            padding: 6px
        }
        QPushButton:pressed{
            background-color: #5B7065
        }
        '''
        self.db = Database('usersdata.db')
        self.save_username = self.username.text()
        self.save_password = self.password.text()
        self.save_repeat_password = self.repeat_password.text()
        if self.save_username != '' and self.save_password != '' and self.save_repeat_password != '':
            if len(self.save_username) <= 12 and len(self.save_username) >= 2 and len(self.save_password) <= 20 and len(self.save_password) >= 5:
                if self.save_username != self.save_password:
                    if self.save_password == self.save_repeat_password:
                        if not self.db.check_user(username=self.save_username):
                            self.today = datetime.datetime.today()
                            self.date = self.today.strftime("%m/%d/%Y")
                            self.db.add_userdata(username=self.save_username, password=self.save_password, join_date=self.date)
                            self.success = QMessageBox()
                            self.success.setWindowTitle('Alert')
                            self.success.setStyleSheet(self.stylesheet)
                            self.success.move(1200, 300)
                            self.success.setIcon(QMessageBox.Information)
                            self.success.setText('Successfully registered!\nGo to the log in window?')
                            self.success.setWindowFlags(Qt.WindowStaysOnTopHint)
                            self.success.setDefaultButton(QMessageBox.Ok)
                            self.success.buttonClicked.connect(self.login_click)
                            self.success.exec_()
                        else:
                            self.reg_error = QMessageBox()
                            self.reg_error.setWindowTitle('Alert')
                            self.reg_error.setStyleSheet(self.stylesheet)
                            self.reg_error.move(1200, 300)
                            self.reg_error.setIcon(QMessageBox.Information)
                            self.reg_error.setText('You\'re already registered\nGo to the log in window?')
                            self.reg_error.setWindowFlags(Qt.WindowStaysOnTopHint)
                            self.reg_error.setDefaultButton(QMessageBox.Ok)
                            self.reg_error.buttonClicked.connect(self.login_click)
                            self.reg_error.exec_()
                    else:
                        self.pass_error = QMessageBox()
                        self.pass_error.setWindowTitle('Alert')
                        self.pass_error.setStyleSheet(self.stylesheet)
                        self.pass_error.move(1200, 300)
                        self.pass_error.setIcon(QMessageBox.Information)
                        self.pass_error.setText('Passwords don\'t match\nTry again?')
                        self.pass_error.setWindowFlags(Qt.WindowStaysOnTopHint)
                        self.pass_error.setDefaultButton(QMessageBox.Ok)
                        self.pass_error.buttonClicked.connect(self.again_click)
                        self.pass_error.exec_()
                else:
                    self.data_error = QMessageBox()
                    self.data_error.setWindowTitle('Alert')
                    self.data_error.setStyleSheet(self.stylesheet)
                    self.data_error.move(1200, 300)
                    self.data_error.setIcon(QMessageBox.Information)
                    self.data_error.setText('Password can\'t be same with the username\nTry again?')
                    self.data_error.setWindowFlags(Qt.WindowStaysOnTopHint)
                    self.data_error.setDefaultButton(QMessageBox.Ok)
                    self.data_error.buttonClicked.connect(self.again_click)
                    self.data_error.exec_()
            else:
                self.characters_error = QMessageBox()
                self.characters_error.setWindowTitle('Alert')
                self.characters_error.setStyleSheet(self.stylesheet)
                self.characters_error.move(1200, 300)
                self.characters_error.setIcon(QMessageBox.Information)
                self.characters_error.setText('Invalid username or password given\nTry again?')
                self.characters_error.setWindowFlags(Qt.WindowStaysOnTopHint)
                self.characters_error.setDefaultButton(QMessageBox.Ok)
                self.characters_error.setDetailedText('Username must contain at least 2 characters and no more than '
                '12 characters\nPassword must contain at least 5 characters and no more than 20 characters'
                )
                self.characters_error.buttonClicked.connect(self.again_click)
                self.characters_error.exec_()
        else:
            self.lines_error = QMessageBox()
            self.lines_error.setWindowTitle('Alert')
            self.lines_error.setStyleSheet(self.stylesheet)
            self.lines_error.move(1200, 300)
            self.lines_error.setIcon(QMessageBox.Information)
            self.lines_error.setText('Not all lines are filled in\nTry again?')
            self.lines_error.setWindowFlags(Qt.WindowStaysOnTopHint)
            self.lines_error.setDefaultButton(QMessageBox.Ok)
            self.lines_error.buttonClicked.connect(self.again_click)
            self.lines_error.exec_()

    def again_click(self):
        self.hide()
        self.show()

    def login_click(self):
        self.hide()
        self.lw = login_win()
        self.lw.show()

code/last_window.py
from profile_window import *
from PyQt5.QtCore import Qt
from PyQt5.QtWidgets import QWidget, QLabel, QVBoxLayout, QPushButton
from PyQt5 import QtGui


class last_win(QWidget):
    def __init__(self):
        super().__init__()
        self.appear()
        self.UI()
        self.connections()

    def appear(self):
        self.setWindowTitle('Last')
        self.setWindowFlags(Qt.WindowStaysOnTopHint)
        self.setStyleSheet('''
        QWidget{
            background-color: #04202C
        }
        QLabel{
            color: #C9D1C8;
            font: bold 20px
        }
        QPushButton{
            color: #C9D1C8;
            font: bold 14px;
            border-style: outset;
            border-width: 2px;
            border-radius: 10px;
            border-color: #5B7065;
            font: bold 14px;
            min-width: 10em;
            padding: 6px
        }
        QPushButton:pressed{
            background-color: #5B7065
        }
        '''
        )
        self.resize(500, 250)
        self.move(400, 300)
        self.setWindowIcon(QtGui.QIcon('logo.png'))

    def UI(self):
        self.welcome_text = QLabel('Last window')
        self.profile_button = QPushButton('My profile')

        self.line = QVBoxLayout()
        self.line.addWidget(self.welcome_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.profile_button, alignment=Qt.AlignCenter)
        self.setLayout(self.line)

    def connections(self):
        self.profile_button.clicked.connect(self.profile_click)

    def profile_click(self):
        self.hide()
        self.pw = profile_win()
        self.pw.show()

code/first_window.py
from join_window import *
from login_window import *
from profile_window import *
from delete_window import *
from PyQt5.QtCore import Qt
from PyQt5.QtWidgets import QApplication, QWidget, QLabel, QVBoxLayout, QPushButton, QHBoxLayout
from PyQt5 import QtGui


class first_win(QWidget):
    def __init__(self):
        super().__init__()
        self.appear()
        self.UI()
        self.connections()

    def appear(self):
        self.setWindowTitle('Welcome')
        self.setWindowFlags(Qt.WindowStaysOnTopHint)
        self.setStyleSheet('''
        QWidget{
            background-color: #04202C
        }
        QLabel{
            color: #C9D1C8;
            font: bold 20px
        }
        QPushButton{
            color: #C9D1C8;
            font: bold 14px;
            border-style: outset;
            border-width: 2px;
            border-radius: 10px;
            border-color: #5B7065;
            font: bold 14px;
            min-width: 10em;
            padding: 6px
        }
        QPushButton:pressed{
            background-color: #5B7065
        }
        '''
        )
        self.resize(500, 250)
        self.move(400, 300)
        #self.setWindowIcon(QtGui.QIcon('logo.png')) - optional

    def UI(self):
        self.text = QLabel('Welcome')
        self.join_button = QPushButton('Join')
        self.login_button = QPushButton('Log in')
        self.profile_button = QPushButton('My profile')
        self.delete_button = QPushButton('Delete account')

        self.line = QVBoxLayout()
        self.line.addWidget(self.text, alignment=Qt.AlignCenter)
        
        self.line2 = QHBoxLayout()
        self.line2.addWidget(self.join_button, alignment=Qt.AlignCenter)
        self.line2.addWidget(self.login_button, alignment=Qt.AlignCenter)

        self.line3 = QHBoxLayout()
        self.line3.addWidget(self.profile_button, alignment=Qt.AlignCenter)
        self.line3.addWidget(self.delete_button, alignment=Qt.AlignCenter)
        self.line.addLayout(self.line2)
        self.line.addLayout(self.line3)
        self.setLayout(self.line)

    def connections(self):
        self.join_button.clicked.connect(self.join_click)
        self.login_button.clicked.connect(self.login_click)
        self.profile_button.clicked.connect(self.profile_click)
        self.delete_button.clicked.connect(self.delete_click)

    def join_click(self):
        self.close()
        self.jw = join_win()
        self.jw.show()

    def login_click(self):
        self.close()
        self.lw = login_win()
        self.lw.show()

    def profile_click(self):
        self.close()
        self.pw = profile_win()
        self.pw.show()

    def delete_click(self):
        self.close()
        self.dw = delete_win()
        self.dw.show()


if __name__ == '__main__':
    app = QApplication([])
    mw = first_win()
    mw.show()
    app.exec_()

code/login_window.py
from last_window import *
from PyQt5.QtCore import Qt
from PyQt5.QtWidgets import QWidget, QLabel, QVBoxLayout, QPushButton, QLineEdit, QMessageBox
from PyQt5 import QtGui
from db import Database


class login_win(QWidget):
    def __init__(self):
        super().__init__()
        self.appear()
        self.UI()
        self.connections()

    def appear(self):
        self.setWindowTitle('Log in')
        self.setWindowFlags(Qt.WindowStaysOnTopHint)
        self.setStyleSheet('''
        QWidget{
            background-color: #04202C
        }
        QLabel{
            color: #C9D1C8;
            font: bold 20px
        }
        QPushButton{
            color: #C9D1C8;
            font: bold 14px;
            border-style: outset;
            border-width: 2px;
            border-radius: 10px;
            border-color: #5B7065;
            font: bold 14px;
            min-width: 10em;
            padding: 6px
        }
        QPushButton:pressed{
            background-color: #5B7065
        }
        QLineEdit{
            border: 5px solid #5B7065;
            font: bold 20px;
            color: #C9D1C8
        }
        '''
        )
        self.resize(700, 450)
        self.move(400, 300)
        self.setWindowIcon(QtGui.QIcon('logo.png'))

    def UI(self):
        self.username_text = QLabel('Username:')
        self.username = QLineEdit()
        self.password_text = QLabel('Password:')
        self.password = QLineEdit()
        self.login_button = QPushButton('Log in')

        self.password.setEchoMode(QLineEdit.Password)

        self.line = QVBoxLayout()
        self.line.addWidget(self.username_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.username, alignment=Qt.AlignCenter)
        self.line.addWidget(self.password_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.password, alignment=Qt.AlignCenter)
        self.line.addWidget(self.login_button, alignment=Qt.AlignCenter)
        self.setLayout(self.line)

    def connections(self):
        self.login_button.clicked.connect(self.click)

    def click(self):
        self.stylesheet = '''
        QMessageBox{
            background-color: #04202C
        }
        QLabel{
            color: #C9D1C8;
            font: bold 20px
        }
        QPushButton{
            color: #C9D1C8;
            font: bold 14px;
            border-style: outset;
            border-width: 2px;
            border-radius: 10px;
            border-color: #5B7065;
            font: bold 14px;
            min-width: 10em;
            padding: 6px
        }
        QPushButton:pressed{
            background-color: #5B7065
        }
        '''
        self.db = Database('usersdata.db')
        self.save_username = self.username.text()
        self.save_password = self.password.text()
        if self.save_username != '' and self.save_password != '':
            if not self.db.check_user(username=self.save_username):
                self.user_error = QMessageBox()
                self.user_error.setWindowTitle('Alert')
                self.user_error.setStyleSheet(self.stylesheet)
                self.user_error.move(1200, 300)
                self.user_error.setIcon(QMessageBox.Information)
                self.user_error.setText('User not found\nTry again?')
                self.user_error.setWindowFlags(Qt.WindowStaysOnTopHint)
                self.user_error.setDefaultButton(QMessageBox.Ok)
                self.user_error.buttonClicked.connect(self.again_click)
                self.user_error.exec_()
            else:
                self.check_pass = self.db.get_user_password(username=self.save_username)
                if self.save_password == self.check_pass:
                    self.success = QMessageBox()
                    self.success.setWindowTitle('Alert')
                    self.success.setStyleSheet(self.stylesheet)
                    self.success.move(1200, 300)
                    self.success.setIcon(QMessageBox.Information)
                    self.success.setText('Successfully loged in!\nGo to the next window?')
                    self.success.setWindowFlags(Qt.WindowStaysOnTopHint)
                    self.success.setDefaultButton(QMessageBox.Ok)
                    self.success.buttonClicked.connect(self.continue_click)
                    self.success.exec_()
                else:
                    self.pass_error = QMessageBox()
                    self.pass_error.setWindowTitle('Alert')
                    self.pass_error.setStyleSheet(self.stylesheet)
                    self.pass_error.move(1200, 300)
                    self.pass_error.setIcon(QMessageBox.Information)
                    self.pass_error.setText('Incorrect password\nTry again?')
                    self.pass_error.setWindowFlags(Qt.WindowStaysOnTopHint)
                    self.pass_error.setDefaultButton(QMessageBox.Ok)
                    self.pass_error.buttonClicked.connect(self.again_click)
                    self.pass_error.exec_()
        else:
            self.lines_error = QMessageBox()
            self.lines_error.setWindowTitle('Alert')
            self.lines_error.setStyleSheet(self.stylesheet)
            self.lines_error.move(1200, 300)
            self.lines_error.setIcon(QMessageBox.Information)
            self.lines_error.setText('Not all lines are filled in\nTry again?')
            self.lines_error.setWindowFlags(Qt.WindowStaysOnTopHint)
            self.lines_error.setDefaultButton(QMessageBox.Ok)
            self.lines_error.buttonClicked.connect(self.again_click)
            self.lines_error.exec_()

    def continue_click(self):
        self.hide()
        self.lw = last_win()
        self.lw.show()

    def again_click(self):
        self.hide()
        self.show()

code/profile_window.py
from PyQt5.QtCore import Qt
from PyQt5.QtWidgets import QWidget, QLabel, QVBoxLayout, QPushButton, QLineEdit, QMessageBox
from PyQt5 import QtGui
from db import Database


class profile_win(QWidget):
    def __init__(self):
        super().__init__()
        self.appear()
        self.UI()
        self.connections()

    def appear(self):
        self.setWindowTitle('Profile')
        self.setWindowFlags(Qt.WindowStaysOnTopHint)
        self.setStyleSheet('''
        QWidget{
            background-color: #04202C
        }
        QLabel{
            color: #C9D1C8;
            font: bold 20px
        }
        QPushButton{
            color: #C9D1C8;
            font: bold 14px;
            border-style: outset;
            border-width: 2px;
            border-radius: 10px;
            border-color: #5B7065;
            font: bold 14px;
            min-width: 10em;
            padding: 6px
        }
        QPushButton:pressed{
            background-color: #5B7065
        }
        QLineEdit{
            border: 5px solid #5B7065;
            font: bold 20px;
            color: #C9D1C8
        }
        '''
        )
        self.resize(500, 250)
        self.move(400, 300)
        self.setWindowIcon(QtGui.QIcon('logo.png'))

    def UI(self):
        self.username_text = QLabel('Enter your username to view profile info:')
        self.username = QLineEdit()
        self.view_button = QPushButton('View')

        self.line = QVBoxLayout()
        self.line.addWidget(self.username_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.username, alignment=Qt.AlignCenter)
        self.line.addWidget(self.view_button, alignment=Qt.AlignCenter)
        self.setLayout(self.line)

    def connections(self):
        self.view_button.clicked.connect(self.click)

    def click(self):
        self.stylesheet = '''
        QMessageBox{
            background-color: #04202C
        }
        QLabel{
            color: #C9D1C8;
            font: bold 20px
        }
        QPushButton{
            color: #C9D1C8;
            font: bold 14px;
            border-style: outset;
            border-width: 2px;
            border-radius: 10px;
            border-color: #5B7065;
            font: bold 14px;
            min-width: 10em;
            padding: 6px
        }
        QPushButton:pressed{
            background-color: #5B7065
        }
        '''
        self.db = Database('usersdata.db')
        self.save_username = self.username.text()
        if self.save_username != '':
            if not self.db.check_user(username=self.save_username):
                self.user_error = QMessageBox()
                self.user_error.setWindowTitle('Alert')
                self.user_error.setStyleSheet(self.stylesheet)
                self.user_error.move(1200, 300)
                self.user_error.setIcon(QMessageBox.Information)
                self.user_error.setText('User not found\nTry again?')
                self.user_error.setWindowFlags(Qt.WindowStaysOnTopHint)
                self.user_error.setDefaultButton(QMessageBox.Ok)
                self.user_error.buttonClicked.connect(self.again_click)
                self.user_error.exec_()
            else:
                self.success = QMessageBox()
                self.success.setWindowTitle('Alert')
                self.success.setStyleSheet(self.stylesheet)
                self.success.move(1200, 300)
                self.success.setIcon(QMessageBox.Information)
                self.success.setText('Profile information:'
                '\nUsername: ' + self.save_username +
                '\nJoin date: ' + self.db.get_user_join_date(username=self.save_username)
                )
                self.success.setWindowFlags(Qt.WindowStaysOnTopHint)
                self.success.setDefaultButton(QMessageBox.Ok)
                self.success.buttonClicked.connect(self.again_click)
                self.success.exec_()
        else:
            self.username_error = QMessageBox()
            self.username_error.setWindowTitle('Alert')
            self.username_error.setStyleSheet(self.stylesheet)
            self.username_error.move(1200, 300)
            self.username_error.setIcon(QMessageBox.Information)
            self.username_error.setText('Username wasn\'t entered\nTry again?')
            self.username_error.setWindowFlags(Qt.WindowStaysOnTopHint)
            self.username_error.setDefaultButton(QMessageBox.Ok)
            self.username_error.buttonClicked.connect(self.again_click)
            self.username_error.exec_()

    def again_click(self):
        self.hide()
        self.show()

code/delete_window.py
from join_window import *
from PyQt5.QtCore import Qt, QTime, QTimer
from PyQt5.QtWidgets import QWidget, QLabel, QVBoxLayout, QPushButton, QMessageBox, QLineEdit
from PyQt5 import QtGui
from db import Database


class delete_win(QWidget):
    def __init__(self):
        super().__init__()
        self.appear()
        self.UI()
        self.connections()

    def appear(self):
        self.setWindowTitle('Delete')
        self.setWindowFlags(Qt.WindowStaysOnTopHint)
        self.setStyleSheet('''
        QWidget{
            background-color: #04202C
        }
        QLabel{
            color: #C9D1C8;
            font: bold 20px
        }
        QPushButton{
            color: #C9D1C8;
            font: bold 14px;
            border-style: outset;
            border-width: 2px;
            border-radius: 10px;
            border-color: #5B7065;
            font: bold 14px;
            min-width: 10em;
            padding: 6px
        }
        QPushButton:pressed{
            background-color: #5B7065
        }
        QLineEdit{
            border: 5px solid #5B7065;
            font: bold 20px;
            color: #C9D1C8
        }
        '''
        )
        self.resize(700, 450)
        self.move(400, 300)
        self.setWindowIcon(QtGui.QIcon('logo.png'))

    def UI(self):
        self.time = QTime(0, 0, 30)
        self.time_count = QLabel(self.time.toString("hh:mm:ss"))
        self.timer = QTimer()
        self.timer.timeout.connect(self.timer_start)
        self.timer.start(1000)
        self.timer_text = QLabel('You have 30 seconds to delete your account')
        self.username_text = QLabel('Username:')
        self.username = QLineEdit()
        self.password_text = QLabel('Password:')
        self.password = QLineEdit()
        self.repeat_password_text = QLabel('Repeat password:')
        self.repeat_password = QLineEdit()
        self.delete_button = QPushButton('Delete')

        self.password.setEchoMode(QLineEdit.Password)
        self.repeat_password.setEchoMode(QLineEdit.Password)

        self.line = QVBoxLayout()
        self.line.addWidget(self.timer_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.time_count, alignment=Qt.AlignCenter)
        self.line.addWidget(self.username_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.username, alignment=Qt.AlignCenter)
        self.line.addWidget(self.password_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.password, alignment=Qt.AlignCenter)
        self.line.addWidget(self.repeat_password_text, alignment=Qt.AlignCenter)
        self.line.addWidget(self.repeat_password, alignment=Qt.AlignCenter)
        self.line.addWidget(self.delete_button, alignment=Qt.AlignCenter)
        self.setLayout(self.line)

    def connections(self):
        self.delete_button.clicked.connect(self.click)

    def click(self):
        self.stylesheet = '''
        QMessageBox{
            background-color: #04202C
        }
        QLabel{
            color: #C9D1C8;
            font: bold 20px
        }
        QPushButton{
            color: #C9D1C8;
            font: bold 14px;
            border-style: outset;
            border-width: 2px;
            border-radius: 10px;
            border-color: #5B7065;
            font: bold 14px;
            min-width: 10em;
            padding: 6px
        }
        QPushButton:pressed{
            background-color: #5B7065
        }
        '''
        self.db = Database('usersdata.db')
        self.save_username = self.username.text()
        self.save_password = self.password.text()
        self.save_repeat_password = self.repeat_password.text()
        if self.save_username != '' and self.save_password != '' and self.save_repeat_password != '':
            if self.save_password == self.save_repeat_password:
                if not self.db.check_user(username=self.save_username):
                    self.user_error = QMessageBox()
                    self.user_error.setWindowTitle('Alert')
                    self.user_error.setStyleSheet(self.stylesheet)
                    self.user_error.move(1200, 300)
                    self.user_error.setIcon(QMessageBox.Information)
                    self.user_error.setText('User not found\nTry again?')
                    self.user_error.setWindowFlags(Qt.WindowStaysOnTopHint)
                    self.user_error.setDefaultButton(QMessageBox.Ok)
                    self.user_error.buttonClicked.connect(self.again_click)
                    self.user_error.exec_()
                else:
                    self.check_pass = self.db.get_user_password(username=self.save_username)
                    if self.save_password == self.check_pass:
                        self.success = QMessageBox()
                        self.success.setWindowTitle('Alert')
                        self.success.setStyleSheet(self.stylesheet)
                        self.success.move(1200, 300)
                        self.success.setIcon(QMessageBox.Information)
                        self.success.setText('Are you sure you want to delete your account?')
                        self.success.setWindowFlags(Qt.WindowStaysOnTopHint)
                        self.success.setDefaultButton(QMessageBox.Ok)
                        self.success.buttonClicked.connect(self.delete_click)
                        self.success.exec_()
                    else:
                        self.pass_error = QMessageBox()
                        self.pass_error.setWindowTitle('Alert')
                        self.pass_error.setStyleSheet(self.stylesheet)
                        self.pass_error.move(1200, 300)
                        self.pass_error.setIcon(QMessageBox.Information)
                        self.pass_error.setText('Incorrect password\nTry again?')
                        self.pass_error.setWindowFlags(Qt.WindowStaysOnTopHint)
                        self.pass_error.setDefaultButton(QMessageBox.Ok)
                        self.pass_error.buttonClicked.connect(self.again_click)
                        self.pass_error.exec_()
            else:
                self.match_error = QMessageBox()
                self.match_error.setWindowTitle('Alert')
                self.match_error.setStyleSheet(self.stylesheet)
                self.match_error.move(1200, 300)
                self.match_error.setIcon(QMessageBox.Information)
                self.match_error.setText('Passwords don\'t match\nTry again?')
                self.match_error.setWindowFlags(Qt.WindowStaysOnTopHint)
                self.match_error.setDefaultButton(QMessageBox.Ok)
                self.match_error.buttonClicked.connect(self.again_click)
                self.match_error.exec_()
        else:
            self.lines_error = QMessageBox()
            self.lines_error.setWindowTitle('Alert')
            self.lines_error.setStyleSheet(self.stylesheet)
            self.lines_error.move(1200, 300)
            self.lines_error.setIcon(QMessageBox.Information)
            self.lines_error.setText('Not all lines are filled in\nTry again?')
            self.lines_error.setWindowFlags(Qt.WindowStaysOnTopHint)
            self.lines_error.setDefaultButton(QMessageBox.Ok)
            self.lines_error.buttonClicked.connect(self.again_click)
            self.lines_error.exec_()

    def timer_start(self):
        global time
        self.time = self.time.addSecs(-1)
        self.time_count.setText(self.time.toString('hh:mm:ss'))
        if self.time.toString('hh:mm:ss') == '00:00:00':
            self.timer.stop()
            self.close()

    def again_click(self):
        if self.time.toString('hh:mm:ss') == '00:00:00':
            self.close()
        else:
            self.hide()
            self.show()

    def delete_click(self):
        if self.time.toString('hh:mm:ss') == '00:00:00':
            self.close()
        else:
            self.db.delete_user(username=self.save_username)
            self.success.close()
            self.delete_success = QMessageBox()
            self.delete_success.setWindowTitle('Alert')
            self.delete_success.setStyleSheet(self.stylesheet)
            self.delete_success.move(1200, 300)
            self.delete_success.setIcon(QMessageBox.Information)
            self.delete_success.setText('User deleted successfully')
            self.delete_success.setWindowFlags(Qt.WindowStaysOnTopHint)
            self.delete_success.setDefaultButton(QMessageBox.Ok)
            self.delete_success.buttonClicked.connect(self.delete_success_click)
            self.delete_success.exec_()

    def delete_success_click(self):
        self.close()